mobile/src/main/java/org/openhab/habdroid/ui/WriteTagActivity.java

```diff
@@ -147,35 +156,58 @@ public void onNewIntent(Intent intent) {
 
         try {
             URI sitemapUri = new URI(mSitemapPage);
-            if (!sitemapUri.getPath().startsWith("/rest/sitemaps")) {
+            String longUriToWrite;
+            String shortUriToWrite = "";
+            if (!TextUtils.isEmpty(mItem) && !TextUtils.isEmpty(mState)) {
+                longUriToWrite = "openhabitem://?" + QUERY_PARAMETER_ITEM_NAME + "=" + mItem
+                        + "&" + QUERY_PARAMETER_STATE + "=" + mState
+                        + "&" + QUERY_PARAMETER_MAPPED_STATE + "=" + mMappedState
+                        + "&" + QUERY_PARAMETER_ITEM_LABEL + "=" + mLabel;
+                shortUriToWrite = "openhabitem://?" + QUERY_PARAMETER_ITEM_NAME + "=" + mItem
+                        + "&" + QUERY_PARAMETER_STATE + "=" + mState;
+            } else if (!sitemapUri.getPath().startsWith("/rest/sitemaps")) {
                 throw new URISyntaxException(mSitemapPage, "Expected a sitemap URL");
+            } else {
+                longUriToWrite = "openhab://" + sitemapUri.getPath().substring(14);
             }
-            StringBuilder uriToWrite = new StringBuilder("openhab://sitemaps");
-            uriToWrite.append(sitemapUri.getPath().substring(14));
-            if (!TextUtils.isEmpty(mItem) && !TextUtils.isEmpty(mCommand)) {
-                uriToWrite.append("?item=").append(mItem).append("&command=").append(mCommand);
-            }
+
             writeTagMessage.setText(R.string.info_write_tag_progress);
-            writeTag(tagFromIntent, uriToWrite.toString());
+            writeTag(tagFromIntent, longUriToWrite, shortUriToWrite);
         } catch (URISyntaxException e) {
             Log.e(TAG, e.getMessage());
             writeTagMessage.setText(R.string.info_write_failed);
         }
     }
 
-    private void writeTag(Tag tag, String uri) {
-        Log.d(TAG, "Creating tag object for URI " + uri);
+    private NdefMessage getNdefMessage(String uri) {
+        if (TextUtils.isEmpty(uri)) {
+            return null;
+        }
+        NdefRecord[] longNdefRecords = new NdefRecord[] { NdefRecord.createUri(uri) };
+        return new NdefMessage(longNdefRecords);
+    }
+
+    private void writeTag(Tag tag, String longUri, String shortUri) {
```

Comment:
This function calls `writeTagMessage.setText()` a few times, but the text is never changed in the UI. This should be fixed by using a AsyncTask.